smslibrary/src/main/java/com/eis/smslibrary/SMSPeer.java

```diff
@@ -4,56 +4,77 @@
 
 import com.eis.communication.Peer;
 import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber;
+
+import java.util.Arrays;
+import java.util.Locale;
 
 /**
  * Implementation of {@link Peer} for SMS communication channel.
  *
  * @author Luca Crema, Marco Mariotto. Reviewed by Marco Cognolato and Mattia Fanan
+ * @author Matteo Carnelos
  */
-public class SMSPeer implements Peer<String>, Comparable<SMSPeer> {
-
-    public static final int MAX_TELEPHONE_NUMBER_LENGTH = 15;
-    public static final int MIN_TELEPHONE_NUMBER_LENGTH = 3;
+public class SMSPeer implements Peer<String>, Comparable<SMSPeer>, java.io.Serializable {
 
+    private static String defaultRegion = Locale.getDefault().getCountry();
     private String telephoneNumber;
 
     /**
-     * @param telephoneNumber Address for the peer.
-     * @throws InvalidTelephoneNumberException If telephoneNumber check is not {@link TelephoneNumberState#TELEPHONE_NUMBER_VALID}.
+     * Set the default country code used when a telephone number without prefix is given.
+     *
+     * @param defaultRegion The default 2 digit ISO country code.
+     * @throws IllegalArgumentException If the given code is not a valid ISO country code.
+     * @author Matteo Carnelos
      */
-    public SMSPeer(String telephoneNumber) throws InvalidTelephoneNumberException {
-        TelephoneNumberState telephoneNumberState = SMSPeer.checkPhoneNumber(telephoneNumber);
-        if(telephoneNumberState != TelephoneNumberState.TELEPHONE_NUMBER_VALID)
-            throw new InvalidTelephoneNumberException(telephoneNumberState);
+    public static void setDefaultRegion(@NonNull String defaultRegion) {
+        if(!Arrays.asList(Locale.getISOCountries()).contains(defaultRegion))
+            throw new IllegalArgumentException("The given string is not a valid ISO country code");
+        SMSPeer.defaultRegion = defaultRegion;
+    }
 
-        this.telephoneNumber = telephoneNumber;
+    /**
+     * Create a new SMSPeer object. If the given number does not explicitly contains the prefix
+     * will be added the default one.
+     * Note: It is not always possible to obtain the correct default country code, especially if the
+     * phone is moving between countries. To set a custom one use the method {@link #setDefaultRegion(String)}.
+     *
+     * @param telephoneNumber Address for the peer.
+     * @throws InvalidTelephoneNumberException If telephoneNumber is not valid.
+     * @throws IllegalArgumentException If the given telephoneNumber is not parsable.
+     * @author Matteo Carnelos
+     */
+    public SMSPeer(@NonNull String telephoneNumber) throws InvalidTelephoneNumberException {
+        if(defaultRegion.isEmpty() && (!telephoneNumber.startsWith("00") || !telephoneNumber.startsWith("+")))
```

Comment:
maybe you mean `if(defaultRegion.isEmpty() && ((!telephoneNumber.startsWith("00") || !telephoneNumber.startsWith("+"))))`?